frontend/src/pages/TermPlanner/ValidateTermPlanner.js

```diff
@@ -25,6 +25,31 @@ const validateTermPlanner = (payload) => {
   };
 };
 
+// takes in startYear (int) and gets current date to return the most recent term that has past (week 2)
+// string of year in degree (not array index) and term most recent is returned e.g. 'Y3T0'
+const getMostRecentPastTerm = (startYear) => {
+  let currTime = new Date();
+  let currYear = currTime.getFullYear();
+  let currMonth = currTime.getMonth();
+  let currDay = currTime.getDate();
+
+  const currDegreeYear = currYear - startYear + 1;
+
+  // session dates gathered from: https://www.student.unsw.edu.au/teaching-periods
+  let lastTermPast;
+  if ((currDay > 29 && currMonth == 8) || currMonth > 8) {
```

Comment:
these should *definitely* be pushed to a highly visible constants file.

Reply:
ohh yes true, will do that